src/Entity/EntityMeta.php

```diff
@@ -153,4 +155,58 @@ public static function baseFieldDefinitions(EntityTypeInterface $entity_type): a
     return $fields;
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function preSave(EntityStorageInterface $storage) {
+    $emptyEntity = TRUE;
+
+    // Avoids to save new revision if no change required.
+    $fields = array_keys($this->toArray());
+    $field_blacklist = $this->traitGetFieldsToSkipFromTranslationChangesCheck($this);
```

Comment:
Nit: we can use lowerCamelCase or snake_case for variable names but they shouldn't be mixed in the same file.